lib/os_io.h

```diff
@@ -93,39 +93,25 @@ typedef struct ___device_struct
 #define MAX_CONDVARS 8192
 #endif
 
-#ifdef USE_poll
-#ifndef MAX_POLLFDS
-#define MAX_POLLFDS MAX_CONDVARS
-#endif
+#ifdef USE_select_or_poll
 
 typedef ___SIZE_TS ___fdbits;
```

Comment:
I don't see the reason why fdbits are of type ```___SIZE_TS```. I would understand ```___UWORD``` which is a wide unsigned integer type, and you can use ```___WORD_WIDTH``` as the number of bits in that type, and ```((fd) >> ___LOG_WORD_WIDTH)``` to implement ```___FD_ELT(fd)```.